### DISCUSSION TITLE
Searching a file across SharePoint sites

### DISCUSSION THREAD

**Discussion by sabrinadeveloper2023** [pnp/pnpcore#1166] [Q&A]
Hi,

Is there a way using PnP Core to search for a document across all our SharePoint sites. All I know is the DocumentID but I dont know in which site is stored.

Thanks

**Comment by jansenbe** [comment]
@sabrinadeveloper2023 : See https://pnp.github.io/pnpcore/using-the-sdk/search-intro.html#making-a-basic-search-request for using search, `DocId` is an OOB managed property that you can use:

![image](https://user-images.githubusercontent.com/7451219/234209157-6b6b5dd1-1cc2-499e-9fc2-e09da9a1f027.png)
